instrumentation/opentelemetry-instrumentation-pymssql/src/opentelemetry/instrumentation/pymssql/__init__.py

```diff
@@ -0,0 +1,198 @@
+    from opentelemetry.instrumentation.pymssql import PyMSSQLInstrumentor
+
+    PyMSSQLInstrumentor().instrument()
+
+    cnx = pymssql.connect(database="MSSQL_Database")
+    cursor = cnx.cursor()
+    cursor.execute("INSERT INTO test (testField) VALUES (123)"
+    cnx.commit()
+    cursor.close()
+    cnx.close()
+
+.. code:: python
+
+    import pymssql
+    from opentelemetry.instrumentation.pymssql import PyMSSQLInstrumentor
+
+    # Alternatively, use instrument_connection for an individual connection
+    cnx = pymssql.connect(database="MSSQL_Database")
+    instrumented_cnx = PyMSSQLInstrumentor().instrument_connection(cnx)
+    cursor = instrumented_cnx.cursor()
+    cursor.execute("INSERT INTO test (testField) VALUES (123)"
+    instrumented_cnx.commit()
+    cursor.close()
+    instrumented_cnx.close()
+
+API
+---
+The `instrument` method accepts the following keyword args:
+
+tracer_provider (TracerProvider) - an optional tracer provider
+"""
+
+from __future__ import annotations
+
+from typing import Any, Callable, Collection, NamedTuple
+
+import pymssql
+
+from opentelemetry.instrumentation import dbapi
+from opentelemetry.instrumentation.instrumentor import BaseInstrumentor
+from opentelemetry.instrumentation.pymssql.package import _instruments
+from opentelemetry.instrumentation.pymssql.version import __version__
+
+_DATABASE_SYSTEM = "mssql"
+
+
+class _PyMSSQLConnectMethodArgsTuple(NamedTuple):
+    server: str | None = None
+    user: str | None = None
+    password: str | None = None
+    database: str | None = None
+    timeout: int | None = None
+    login_timeout: int | None = None
+    charset: str | None = None
+    as_dict: bool | None = None
+    host: str | None = None
+    appname: str | None = None
+    port: str | None = None
+    conn_properties: str | None = None
+    autocommit: bool | None = None
+    tds_version: str | None = None
+
+
+class _PyMSSQLDatabaseApiIntegration(dbapi.DatabaseApiIntegration):
+    def wrapped_connection(
+        self,
+        connect_method: Callable[..., Any],
+        args: tuple[Any, Any],
+        kwargs: dict[Any, Any],
+    ):
+        """Add object proxy to connection object."""
+        connection = connect_method(*args, **kwargs)
+        connect_method_args = _PyMSSQLConnectMethodArgsTuple(*args)
+
+        self.name = self.database_system
+        self.database = kwargs.get("database") or connect_method_args.database
+
+        user = kwargs.get("user") or connect_method_args.user
+        if user is not None:
+            self.span_attributes["db.user"] = user
+
+        port = kwargs.get("port") or connect_method_args.port
+        host = kwargs.get("server") or connect_method_args.server
+        if host is None:
+            host = kwargs.get("host") or connect_method_args.host
+            if host is not None:
+                # The host string can include the port, separated by either a coma or
+                # a column
+                for sep in (":", ","):
```

Comment:
Could we also have some test coverage for setting the span attributes, especially for this custom creation of `host`?

Reply:
Done!